packages/headless/src/composables/helpers/useOutsideClick.ts

```diff
@@ -22,19 +28,42 @@ export type UseOutsideClickOptions = {
  * Composable for listening to click events that occur outside of a component.
  * Useful to e.g. close flyouts or tooltips.
  */
-export const useOutsideClick = ({ inside, onOutsideClick, disabled }: UseOutsideClickOptions) => {
+export const useOutsideClick = <TCheckOnTab extends boolean | undefined>({
+  inside,
+  onOutsideClick,
+  disabled,
+  checkOnTab,
+}: UseOutsideClickOptions<TCheckOnTab>) => {
+  const isOutsideClick = (target: EventTarget | null) => {
+    if (!target) return true;
+    const raw = toValue(inside);
+    const elements = Array.isArray(raw) ? raw : [raw];
+    return !elements.some((element) => element?.contains(target as Node | null));
+  };
+
   /**
    * Document click handle that closes then tooltip when clicked outside.
    * Should only be called when trigger is "click".
    */
-  const listener = (event: MouseEvent) => {
-    const raw = toValue(inside);
-    const elements = Array.isArray(raw) ? raw : [raw];
-    const isOutsideClick = !elements.some((element) =>
-      element?.contains(event.target as HTMLElement),
-    );
-    if (isOutsideClick) onOutsideClick(event);
+  const clickListener = (event: MouseEvent) => {
+    if (isOutsideClick(event.target)) onOutsideClick(event);
   };
 
-  useGlobalEventListener({ type: "click", listener, disabled });
+  useGlobalEventListener({ type: "click", listener: clickListener, disabled });
+
+  if (checkOnTab) {
+    const keydownListener = async (event: KeyboardEvent) => {
+      if (event.key !== "Tab") return;
+
+      // using setTimeout here to guarantee that side effects that might change the document.activeElement have run before checking
+      // the activeElement
+      await new Promise((resolve) => setTimeout(resolve));
+
+      if (isOutsideClick(document.activeElement)) {
+        onOutsideClick(event as Parameters<typeof onOutsideClick>[0]);
+      }
+    };
+
+    useGlobalEventListener({ type: "keydown", listener: keydownListener, disabled });
+  }
```

Comment:
Would prefer to use the focusout (or similar) event, so that we don't have a need for `setTimeout`

```suggestion
  if (checkOnTab) {
    const focusoutListener = async (event: FocusEvent) => {
      // check if the new focus target is inside or outside
      if (inside.every(el => !el.contains(event.relatedTarget))) {
        onOutsideClick(event as Parameters<typeof onOutsideClick>[0]);
      }
    };

    useGlobalEventListener({ type: "focusout", listener: focusoutListener, disabled });
  }
```

Reply:
As discussed, I created a follow up ticket for this so this PR can be merged: #3633